torch/distributions/pareto.py

```diff
@@ -0,0 +1,45 @@
+from numbers import Number
+
+import math
+
+import torch
+from torch.distributions.distribution import Distribution
+from torch.distributions.utils import broadcast_all
+
+
+class Pareto(Distribution):
+    r"""
+    Samples from a Pareto Type 1 distribution.
+
+    Example::
+
+        >>> m = Pareto(torch.Tensor([1.0]), torch.Tensor([1.0]))
+        >>> m.sample() # sample from a Pareto distribution with scale=1 and alpha=1
+         1.5623
+        [torch.FloatTensor of size 1]
+
+    Args:
+        scale (float or Tensor or Variable): Scale parameter of the distribution
+        alpha (float or Tensor or Variable): Shape parameter of the distribution
+    """
+    has_rsample = True
+
+    def __init__(self, scale, alpha):
+        self.scale, self.alpha = broadcast_all(scale, alpha)
+        if isinstance(scale, Number) and isinstance(alpha, Number):
+            batch_shape = torch.Size()
+        else:
+            batch_shape = self.scale.size()
+        super(Pareto, self).__init__(batch_shape)
+
+    def rsample(self, sample_shape=torch.Size()):
+        shape = self._extended_shape(sample_shape)
+        exp_dist = self.alpha.new(shape).exponential_()
+        return self.scale.new(shape) * torch.exp(exp_dist)
```

Comment:
I think this should be
```diff
- return self.scale.new(shape) * torch.exp(exp_dist)
+ return self.scale * torch.exp(exp_dist)
```
I believe `self.scale.new(shape)` returns a new tensor filled with uninitialized memory. The `new()` method is ok to use above since we immediately fill that uninitialized memory with `.exponential_()` values.

I'm concerned that the `test_pareto_sample()` tests did not catch this. Have you seen any local test failures?